### PR TITLE
Introduce protopipe

### PR DESCRIPTION
**Pull request by Maxim-Doronin**
### Details:
 - Publishing protopipe to open-source 

### Tickets:
 - E-143100

### CODE REVIEW

File: src/plugins/intel_npu/thirdparty/CMakeLists.txt [hunk 1]
```diff
@@ -12,3 +12,15 @@ if(ENABLE_ZEROAPI_BACKEND)
...
 endif()
+
+#
+# yaml-cpp
+#
+
+if(ENABLE_INTEL_NPU_PROTOPIPE)
+    add_subdirectory(yaml-cpp EXCLUDE_FROM_ALL)
+    # NB: Suppress warnings in yaml-cpp
+    if(NOT MSVC)
```
Comment (ilya-lavrenov) [inline src/plugins/intel_npu/thirdparty/CMakeLists.txt:23]:
```suggestion
    if(SUGGEST_OVERRIDE_SUPPORTED)
```
as this option is not supported by all non MSVC compilers.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ArtemySkrebkov — APPROVED** [review]
I reviewed the cmake changes 👍 

@TolyaTalamanov 

I think it makes sense to get a review from you for this PR

**Comment by Maxim-Doronin** [comment]
> I reviewed the cmake changes 👍
> 
> @TolyaTalamanov
> 
> I think it makes sense to get a review from you for this PR

FYI, I haven't changed anything in c++ except for copyright headers. All changes are in cmake

File: src/plugins/intel_npu/thirdparty/CMakeLists.txt (hunk 1 above)
Comment (Maxim-Doronin) [reply src/plugins/intel_npu/thirdparty/CMakeLists.txt]: Thanks, fixed in https://github.com/openvinotoolkit/openvino/pull/27087/commits/369948f45b37a5f547a7ae4a6ed2e951a6637098

**Review by PatrikStepan — APPROVED** [review]

**Pull request merged**